components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowConsumer.java

```diff
@@ -181,6 +181,14 @@ public void handleRequest(HttpServerExchange httpExchange) throws Exception {
     private void sendResponse(HttpServerExchange httpExchange, Exchange camelExchange) throws IOException, NoTypeConversionAvailableException {
         Object body = getResponseBody(httpExchange, camelExchange);
 
+        Message answer;
+        if (camelExchange.hasOut()) {
```

Comment:
I think `hasOut` is being deprecated here https://github.com/apache/camel/blob/master/core/camel-api/src/main/java/org/apache/camel/Exchange.java#L456, I think is better to use `getMessage` instead

Reply:
thanks, saw that it was deprecated but also is used in a lot of places. Was not sure what the preferred approach was.

also, was going to clean this all up a bit. mostly wanted to know if this was the right place to but the code since it was different than Netty-Http (see #3219 )

for some failed tests, that currently have no body but return 200, what would be the preferred approach to fixing. Add a body or change the expected status to 204?

Reply:
IMHO having 204 is more clear natural that corresponds to the status code definition